### ISSUE TITLE
Underlined toggle button height

### ISSUE THREAD

**Issue by Chrriis** [JFormDesigner/FlatLaf#44]
I want to use toggle buttons like tabs so I set the underline type. I also increase the size of the underline. The problem is that the line is too close from the text and I don't know if there is a way to have a taller button when the style is underline (something similar to the "TabbedPane.tabHeight" property).

Here is an example comparing tabs and toggle buttons:
```Java
	public static void main(String[] args) throws Exception {
		UIManager.setLookAndFeel(new FlatLightLaf());
		UIDefaults defaults = UIManager.getDefaults();
		defaults.put("ToggleButton.underline.underlineHeight", 3);
		defaults.put("TabbedPane.tabSelectionHeight", 3);
		defaults.put("TabbedPane.tabHeight", 22);
		JFrame frame = new JFrame();
		frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		JPanel centerPane = new JPanel(new BorderLayout());
		JToolBar toolBar = new JToolBar();
		toolBar.setFloatable(false);
		ButtonGroup buttonGroup = new ButtonGroup();
		JToggleButton button1 = new JToggleButton("My first button");
		button1.setSelected(true);
		button1.putClientProperty(FlatClientProperties.BUTTON_TYPE, FlatClientProperties.BUTTON_TYPE_UNDERLINE);
		buttonGroup.add(button1);
		toolBar.add(button1);
		JToggleButton button2 = new JToggleButton("My second button");
		button2.putClientProperty(FlatClientProperties.BUTTON_TYPE, FlatClientProperties.BUTTON_TYPE_UNDERLINE);
		buttonGroup.add(button2);
		toolBar.add(button2);
		centerPane.add(toolBar, BorderLayout.NORTH);
		JTabbedPane tabbedPane = new JTabbedPane();
		tabbedPane.setTabLayoutPolicy(JTabbedPane.SCROLL_TAB_LAYOUT);
		tabbedPane.addTab("My first tab", new JPanel());
		tabbedPane.addTab("My second tab", new JPanel());
		centerPane.add(tabbedPane);
		frame.getContentPane().add(centerPane);
		frame.setSize(400, 300);
		frame.setVisible(true);
	}

```

... and the result:
![UnderlineToggleButtonsVsTabs](https://user-images.githubusercontent.com/1174202/72332724-d430b400-36ba-11ea-9617-740cc90c9f94.png)

**Comment by DevCharly** [comment]
Setting minimum height with client property `JComponent.minimumHeight` (or `FlatClientProperties.MINIMUM_HEIGHT`) is now implemented in master branch.

Do you think that a global value in the UI defaults is also necessary/useful?

**Comment by Chrriis** [comment]
I don't think it is necessary, and I don't know if it could be useful. Because underline is not the default style for a button, it requires code to activate it, so if someone wants a different height, they could make a factory method.

**Comment by Chrriis** [comment]
I just tried the code, but I would have thought the preferred size is the one that needs attention. How would you change my test case so that the button is taller (that means having the underline lower)?

**Comment by DevCharly** [comment]
Must admit that I've not tested the change with your code.

It does not work because the JToggleButton is a child of a JToolBar and such children do not get "minimum preferred" width/height.
https://github.com/JFormDesigner/FlatLaf/blob/74909da110f81ee4eb6a67a5504f9daa8e5ca529/flatlaf-core/src/main/java/com/formdev/flatlaf/ui/FlatButtonUI.java#L387-L391

The purpose of the "minimum preferred" width is that buttons with short text (e.g. "OK") or empty text fields have a useful width by default. If a button is in a toolbar, it should be smaller.

Anyway there is a better way to give the button a larger size:
~~~java
button1.setMargin( new Insets( 10, 10, 10, 10 ) );
~~~

**Comment by DevCharly** [comment]
implemented in 0.26
